packages/openid4vc/src/openid4vc-verifier/OpenId4VcSiopVerifierService.ts

```diff
@@ -608,9 +624,28 @@ export class OpenId4VcSiopVerifierService {
 
         let isValid: boolean
 
-        // TODO: it might be better here to look at the presentation submission to know
-        // If presentation includes a ~, we assume it's an SD-JWT-VC
-        if (typeof encodedPresentation === 'string' && encodedPresentation.includes('~')) {
+        if (typeof encodedPresentation === 'string' && MdocDeviceResponse.isBase64DeviceResponse(encodedPresentation)) {
+          if (!options.responseUri || !options.mdocGeneratedNonce) {
+            isValid = false
```

Comment:
let's add a reason to the return value. as this is very hard to debug that you didn't provide these values: https://github.com/Sphereon-Opensource/OID4VC/blob/3b516bef69484cf7af370ea0213e486f4f701d28/packages/siop-oid4vp/lib/authorization-response/types.ts#L94